apps/desktop/src/lib/state/customHooks.svelte.ts

```
@@ -217,24 +270,25 @@ export function buildMutationHooks<Definitions extends EndpointDefinitions>({
 	 * Replicate the behavior of `useMutation` from RTK Query.
 	 * @see: https://github.com/reduxjs/redux-toolkit/blob/637b0cad2b227079ccd0c5a3073c09ace6d8759e/packages/toolkit/src/query/react/buildHooks.ts#L867-L935
 	 */
-	function useMutation(
-		params?: UseMutationHookParams<MutationDefinition<any, any, any, any, any>>
-	) {
+	function useMutation(params?: UseMutationHookParams<D>) {
 		const { fixedCacheKey, preEffect, sideEffect, onError } = params || {};
 		const dispatch = getDispatch();
 
-		let promise =
-			$state<MutationActionCreatorResult<MutationDefinition<unknown, any, any, unknown>>>();
+		let promise = $state<MutationActionCreatorResult<D>>();
 
-		async function triggerMutation(queryArg: unknown) {
+		async function triggerMutation(queryArg: QueryArgFrom<D>) {
 			preEffect?.(queryArg);
-			const dispatchResult = dispatch(initiate(queryArg, { fixedCacheKey }));
-			promise = dispatchResult;
-			const result = await promise.unwrap().catch((error) => {
-				onError?.(error, queryArg);
-			});
```

don't swallow the error